File: core/reasoning/topology_reasoning_engine.py
from __future__ import annotations

from typing import Any, Dict

from core.graph.topology_reasoning_engine import reason_topology
from core.graph.semantic_cycle_analysis_engine import detect_cycles


def reason_topology_semantic(graph: Dict[str, Any]) -> Dict[str, Any]:
    topo = reason_topology(graph)
    cycles = detect_cycles(graph)
    return {**topo, "cycles": cycles, "contradiction_pressure": cycles.get("contradiction_pressure", 0), "explainable": True}

File: src/reasoning/topologyReasoningEngine.ts
/**
 * Converted from Python: core/reasoning/topology_reasoning_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";
import { reasonTopology } from "../graph/topologyReasoningEngine.js";
import { detectCycles } from "../graph/semanticCycleAnalysisEngine.js";

export function reasonTopologySemantic(graph: any): any {
  var topo: any = reasonTopology(graph);
  var cycles: any = detectCycles(graph);
  return {...(topo), "cycles": cycles, "contradiction_pressure": py.get(cycles, "contradiction_pressure", 0), "explainable": true};
}
export { detectCycles, reasonTopology };
